python/ray/tests/test_scheduling.py

```diff
@@ -477,6 +477,29 @@ def put():
     ray.get(tasks, timeout=30)
 
 
+def test_nested_task(ray_start_regular_shared):
+    @ray.remote
+    def outer():
+        ref = inner.remote()
+        # This sleep is necessary due to a race condition because workers are
+        # blocked/unblocked from the main thread, while tasks are submitted
+        # from a dedicated thread. If the worker is blocked before the inner
+        # task is submitted, the raylet will schedule another instance of
+        # `outer` since there are no `inner`'s queued.
+        time.sleep(0.01)
```

Comment:
If we don't sleep here, will it fail? It introduces the uncertainty for the test.

Reply:
Fixed the race condition.